File: ms3-pedidos/src/clients/ms2Client.js
const axios = require("axios");

const MS2_BASE_URL = process.env.MS2_BASE_URL || "http://localhost:8082";

async function getDish(restaurantId, dishId) {
  const { data } = await axios.get(
    `${MS2_BASE_URL}/api/v1/restaurants/${restaurantId}/menu/${dishId}`,
    { timeout: 5000 }
  );
  return data;
}

module.exports = { getDish };

File: ms4-agregador-rastreo/app/clients/ms2_client.py
import os

import httpx

MS2_BASE_URL = os.getenv("MS2_BASE_URL", "http://localhost:8082")


async def get_restaurant(restaurant_id: str):
    async with httpx.AsyncClient(timeout=5.0) as client:
        resp = await client.get(f"{MS2_BASE_URL}/api/v1/restaurants/{restaurant_id}")
        resp.raise_for_status()
        return resp.json()
